src/platforms/ios/iosutils.mm

```diff
@@ -115,11 +116,16 @@
   }
 }
 
-// static
+@interface StatusBarModifierViewController : UIViewController
+@property (nonatomic, assign) UIStatusBarStyle preferredStatusBarStyle;
+@end
+
 void IOSUtils::setStatusBarTextColor(const Theme::StatusBarTextColor color) {
-  if (color == Theme::StatusBarTextColorLight) {
-    [UIApplication sharedApplication].statusBarStyle = UIStatusBarStyleLightContent;
-  } else {
-    [UIApplication sharedApplication].statusBarStyle = UIStatusBarStyleDarkContent;
-  }
+    StatusBarModifierViewController* rootViewController = static_cast<StatusBarModifierViewController *>([[UIApplication sharedApplication].windows[0] rootViewController]);
```

Comment:
Not sure if this static_cast is too much of a hack. This is the most straight forward way I could find to have a setter for `preferredStatusBarStyle`. I am totally new to Objective-C so any input is appreciated :)

Reply:
No, it's actually correct. static_cast happens a compilation time.